go/pkg/tools/tools_test.go

```diff
@@ -35,22 +36,51 @@ var _ = Describe("Tools", func() {
 			block, err := tools.ConfigTxGen(profile, "mychannel")
 			Expect(err).NotTo(HaveOccurred())
 			Expect(block).ToNot(BeNil())
-			envelopeConfig, err := protoutil.ExtractEnvelope(block, 0)
-			Expect(err).NotTo(HaveOccurred())
-			cryptoProvider, err := sw.NewDefaultSecurityLevelWithKeystore(sw.NewDummyKeyStore())
+			_, err = ExtractEnvelope(block, 0)
 			Expect(err).NotTo(HaveOccurred())
-			bundle, err := channelconfig.NewBundleFromEnvelope(envelopeConfig, cryptoProvider)
+			//_, err = sw.NewDefaultSecurityLevelWithKeystore(sw.NewDummyKeyStore())
+			//Expect(err).NotTo(HaveOccurred())
+			/*bundle, err := channelconfig.NewBundleFromEnvelope(envelopeConfig, cryptoProvider)
 			Expect(err).NotTo(HaveOccurred())
 			oc, exists := bundle.OrdererConfig()
 			Expect(exists).To(BeTrue())
 			configMetadata := &etcdraft.ConfigMetadata{}
 			err = proto.Unmarshal(oc.ConsensusMetadata(), configMetadata)
 			Expect(err).NotTo(HaveOccurred())
-			Expect(configMetadata.Options).NotTo(BeNil())
+			Expect(configMetadata.Options).NotTo(BeNil())*/
 		})
 	})
 
 	PIt("gate policy", func() {})
 	PIt("peer discovery", func() {})
 	PIt("generate connection profile for sdk", func() {})
 })
+
+// ExtractEnvelope retrieves the requested envelope from a given block and
+// unmarshals it
+func ExtractEnvelope(block *cb.Block, index int) (*cb.Envelope, error) {
+	if block.Data == nil {
+		return nil, errors.New("block data is nil")
+	}
+
+	envelopeCount := len(block.Data.Data)
+	if index < 0 || index >= envelopeCount {
+		return nil, errors.New("envelope index out of bounds")
```

Comment:
If you are going to the effort of checking the index is in bounds rather than just letting it fail when you index the slice below, I would include more information in the error
```suggestion
		return nil, fmt.Errorf("envelope index %d out of bounds for block containing %d envelopes", index, envelopeCount)
```